buildattr64/buildattr64.rst

```diff
@@ -0,0 +1,904 @@
+presence of ``.ARM.attributes`` sections in a loadable-unit is
+permitted, but the interpretation of the contents of the section is
+outside the scope of this document. The use of ``.note.gnu.property``
+sections for loadable-units is recommended for SysVr4 platforms,
+see (SYSVABI64_) for details.
+
+.. raw:: pdf
+
+   PageBreak
+
+Formal syntax of an ELF Attributes Section
+==========================================
+
+An ELF Attributes section uses a processor specific section with details
+
+.. table:: Attributes Section
+
+    +---------------------+----------------------------+-------+-----------+------+------------+
+    | Name                | Type                       | Flags | Alignment | Info | Entry Size |
+    +=====================+============================+=======+===========+======+============+
+    | ``.ARM.attributes`` | ``SHT_AARCH64_ATTRIBUTES`` | 0x0   | 1         | 0    | 0          |
+    +---------------------+----------------------------+-------+-----------+------+------------+
+
+A consumer may not assume the natural alignment of a data type such as ``uint32``.
+
+The syntactic structure of an attributes section is::
+
+   <format-version: <version-letter>>
+   [ <uint32: subsection-length> NTBS: vendor-name
+     <bytes: vendor-data>
+   ]*
+
+*Format-version* describes the format of the following data. It is a
+single byte. Only version 'A' (0x41) is defined. This field exists to
+allow future changes in format.
+
+*Subsection-length* is a 4-byte integer in the byte order of the ELF
+file. It encodes the length of the subsection, including the length
+field itself, the vendor name string and its terminating NUL byte, and
+the following vendor data. It gives the offset from the start of this
+subsection to the start of the next one.
+
+*Vendor-name* is a NUL-terminated byte string (NTBS) like a C-language
+literal string.
+
+No requirements are placed on the format of private vendor data. The
+format of the public attributes subsections (which will have a
+pseudo-vendor name with a prefix of “aeabi”) are described below.
+
+Attributes that record data about the compatibility of this
+relocatable object file with other relocatable object files must be
+recorded in a public "aeabi" prefixed subsection.
+
+Attributes meaningful only to the producer must be recorded in a
+private vendor subsection. These must not affect compatibility between
+relocatable object files. A producer that does not recognize the
+private vendor may skip over the private vendor subsection.
+
+Formally, there are no constraints on the order or number of vendor
+subsections. A consumer can collect the public ("aeabi" prefixed)
+attributes in a single pass over the section, then all of its private
+data in a second pass.
+
+Public aeabi prefixed subsections
+=================================
+
+About public tags
+-----------------
+
+Subsections containing a vendor name with a prefix of "aeabi" are
```

Comment:
Subsections containing a vendor name with a prefix of "aeabi"

"containing vendor name"
If it is correct to understand that aeabi-feature-and-bits and aeabi-pauthabi are some of those "public subsections" that starts with aeabi mentioned here, then where is the "vendor name"? Isn't vendor name supposed to be something like "Arm", "Intel" etc.?